File: app/src/main/java/pl/tobzzo/miechowskiepowietrze/di/AppComponent.java
package pl.tobzzo.miechowskiepowietrze.di;

import dagger.BindsInstance;
import dagger.Component;
import dagger.android.support.AndroidSupportInjectionModule;
import javax.inject.Singleton;
import pl.tobzzo.miechowskiepowietrze.MpowApplication;
//import pl.tobzzo.miechowskiepowietrze.activities.MainActivityWithCharts;
//import pl.tobzzo.miechowskiepowietrze.analytics.MpowAnalyticsComponent;
//import pl.tobzzo.miechowskiepowietrze.network.MpowNetworkComponent;
//import pl.tobzzo.miechowskiepowietrze.utils.MpowApiKeyProvider;

@Singleton
@Component(modules = { AndroidSupportInjectionModule.class, AppModule.class, BuilderModule.class })
public interface AppComponent {

  @Component.Builder
  interface Builder {

    @BindsInstance
    Builder application(MpowApplication application);
    AppComponent build();
  }

  void inject(MpowApplication app);

  //MainComponent plus(MainModule mainModule);
  //
  //SensorComponent plus(SensorModule sensorModule);

  //void inject(MpowNetworkComponent mpowNetworkComponent);
  //
  //void inject(MpowAnalyticsComponent mpowAnalyticsComponent);
  //
  //void inject(MainActivityWithCharts mainActivityWithCharts);
  //
  //void inject(MpowApiKeyProvider mpowApiKeyProvider);
}

File: app/src/main/java/pl/tobzzo/miechowskiepowietrze/di/AppComponent.kt
package pl.tobzzo.miechowskiepowietrze.di

import android.app.Application
import dagger.BindsInstance
import dagger.Component
import dagger.android.AndroidInjector
import dagger.android.DaggerApplication
import dagger.android.support.AndroidSupportInjectionModule
import pl.tobzzo.miechowskiepowietrze.app.MpowApplication
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(ApplicationModule::class,
        ActivityBindingModule::class,
        AndroidSupportInjectionModule::class
        ))

interface AppComponent : AndroidInjector<MpowApplication> {

    override fun inject(instance: MpowApplication)

    @Component.Builder
    interface Builder {

        @BindsInstance
        fun application(application: MpowApplication): Builder

        fun build(): AppComponent
    }
}
